### PR TITLE
Add member stats

### PR DESCRIPTION
**Pull request by ivanhrabcak**
Adds a new stat (time vs members in the discord server):
The graph always starts at the date of creation of the guild.
image

### CODE REVIEW

File: src/stats/show_stats.py [hunk 1]
```diff
@@ -69,11 +73,43 @@ async def show_stats(ctx, stat_id):
...
+        for member in ctx.guild.members:
+            if not member.bot:
+                join_dates.append(member.joined_at)
+
+        join_dates.sort(key=lambda x: time.mktime(x.timetuple()))
+
+        server_created = ctx.guild.created_at
+        existing_for = dt.now(timezone.utc) - server_created
+
+        step = existing_for / 10
```
Comment (xjahic) [inline src/stats/show_stats.py:88]: Please add a comment what is this line for, what it does

File: src/stats/show_stats.py [hunk 2]
```diff
@@ -125,6 +161,23 @@ def make_percentage_graph(stat_id, options, answers, title, graph_size=None):
     plt.clf()
 
 
+def make_time_line_plot(stat_id, data_points, data_values, title, graph_size=None):
```
Comment (xjahic) [inline src/stats/show_stats.py:164]: `make_timeline_plot`

**Review by xjahic — APPROVED** [review]
Though I don't know how the algorithm works, it looks good :).
I have just a small updates and also I have run this stat on my test server and dates on X axis don't look good. Do you maybe know why?

File: src/stats/stat_type.py [hunk 3]
```diff
@@ -1,2 +1,3 @@
 # Use when you want to show many percent each answer has (works only if question has single choice answers)
 PERCENTAGE_BAR = "Percentage bar"
+TIME_LINE_PLOT = "Time line plot"
```
[the remark this answers is not in the document]
Comment (ivanhrabcak) [reply src/stats/stat_type.py]: Renamed in 3214a02.

File: src/stats/available_stats.py [hunk 4]
```diff
@@ -38,6 +38,7 @@
         PERCENTAGE_BAR,
         "Epizódy s hosťami - percentuálny podiel",
     ),
+    "12": (-1, TIME_LINE_PLOT, "Počet ľudí v discord serveri v čase"),
```
[the remark this answers is not in the document]
Comment (ivanhrabcak) [reply src/stats/available_stats.py]: Changed in [3214a02](https://github.com/StreetOfCode/discord-bot/pull/43/commits/3214a02c2989735a84a4664407f1318286ceb138).

File: src/stats/show_stats.py (hunk 1 above)
Comment (ivanhrabcak) [reply src/stats/show_stats.py]: I added 2 comments that explain how the code works ([3214a02](https://github.com/StreetOfCode/discord-bot/pull/43/commits/3214a02c2989735a84a4664407f1318286ceb138)).

File: src/stats/show_stats.py (hunk 2 above)
Comment (ivanhrabcak) [reply src/stats/show_stats.py]: Renamed in [3214a02](https://github.com/StreetOfCode/discord-bot/pull/43/commits/3214a02c2989735a84a4664407f1318286ceb138).

File: src/stats/show_stats.py [hunk 5]
```diff
@@ -69,11 +73,43 @@ async def show_stats(ctx, stat_id):
...
-            db.add_show_stats_cache(stat_id, imgur_url)
-            await send_graph_to_channel(ctx, imgur_url, stat_title)
-        else:
-            await ctx.channel.send("Server preťažený, skús prosím neskôr.")
+    elif stat_type == TIME_LINE_PLOT:
+        join_dates = []
+
+        for member in ctx.guild.members:
+            if not member.bot:
+                join_dates.append(member.joined_at)
```
[the remark this answers is not in the document]
Comment (ivanhrabcak) [reply src/stats/show_stats.py]: It works, didn't think of that! ([3214a02](https://github.com/StreetOfCode/discord-bot/pull/43/commits/3214a02c2989735a84a4664407f1318286ceb138))

File: src/stats/show_stats.py [hunk 6]
```diff
@@ -69,11 +73,43 @@ async def show_stats(ctx, stat_id):
...
+
+        join_dates.sort(key=lambda x: time.mktime(x.timetuple()))
+
+        server_created = ctx.guild.created_at
+        existing_for = dt.now(timezone.utc) - server_created
```
[the remark this answers is not in the document]
Comment (ivanhrabcak) [reply src/stats/show_stats.py]: Renamed in [3214a02](https://github.com/StreetOfCode/discord-bot/pull/43/commits/3214a02c2989735a84a4664407f1318286ceb138).

File: src/stats/show_stats.py [hunk 7]
```diff
@@ -69,11 +73,43 @@ async def show_stats(ctx, stat_id):
         make_percentage_graph(
             stat_id, options, answers, stat_title, graph_size=graph_size
         )
-        if imgur_url := await upload_graph_to_imgur_or_none(stat_id):
-            db.add_show_stats_cache(stat_id, imgur_url)
-            await send_graph_to_channel(ctx, imgur_url, stat_title)
-        else:
-            await ctx.channel.send("Server preťažený, skús prosím neskôr.")
+    elif stat_type == TIME_LINE_PLOT:
```
[the remark this answers is not in the document]
Comment (ivanhrabcak) [reply src/stats/show_stats.py]: > I have just a small updates and also I have run this stat on my test server and dates on X axis don't look good. Do you maybe know why?

I only noticed this issue when I tested it in a server that is older. It is now fixed - at least for me ([3214a02](https://github.com/StreetOfCode/discord-bot/pull/43/commits/3214a02c2989735a84a4664407f1318286ceb138)).

**Comment by ivanhrabcak** [comment]
> I have just a small updates and also I have run this stat on my test server and dates on X axis don't look good. Do you maybe know why?

I only noticed this issue when I tested it in a server that is older. It is now fixed - at least for me ([3214a02](https://github.com/StreetOfCode/discord-bot/pull/43/commits/3214a02c2989735a84a4664407f1318286ceb138)).

**Pull request merged**